pulp_ansible/app/utils.py

```diff
@@ -26,3 +27,26 @@ def filter_content_for_repo_version(qs, repo_version):
     content_rel = RepositoryContent.objects.filter(f)
 
     return qs.filter(pk__in=content_rel.values_list("content_id"))
+
+
+def get_queryset_annotated_with_task_sync_task(qs):
+    tasks = (
+        Task.objects.filter(
+            name__contains="sync", reserved_resources_record__icontains=OuterRef("outer_pk")
```

Comment:
Could you try putting the `OuterRef` inside a `Cast` so that we don't have to annotate the parent queryset to perform the cast on the pk?

Reply:
Yep, I got confused while experimenting with various solutions. Fixed, but still trying to figure out on how to implement prefetch or select_related.